notebook/notebookapp.py

```diff
@@ -2289,14 +2319,16 @@ def list_running_servers(runtime_dir=None):
     if not os.path.isdir(runtime_dir):
         return
 
+    api_version_json_bytes = json.dumps({"version":__version__}).encode()
     for file_name in os.listdir(runtime_dir):
         if re.match('nbserver-(.+).json', file_name):
```

Comment:
In testing, I happened to rename the file to `nbserver-8889.jsonxx` and it still matched.  We should probably make the match a bit more robust.
```suggestion
        if re.match('^nbserver-(.+).json$', file_name):
```

Reply:
nod